android/KMEA/app/src/main/java/com/keyman/engine/util/KMLog.java

```diff
@@ -34,6 +35,44 @@ public static void LogInfo(String tag, String msg) {
     }
   }
 
+  /**
+   * Utility to log info and add as a Sentry breadcrumb, rather than
+   * as an independent message
+   * @param tag String of the caller
+   * @param msg String of the info message
+   */
+  public static void LogBreadcrumb(String tag, String msg, boolean addStackTrace) {
+    if (msg != null && !msg.isEmpty()) {
+      Log.i(tag, msg);
+
+      if (DependencyUtil.libraryExists(LibraryType.SENTRY) && Sentry.isEnabled()) {
+        Breadcrumb crumb = new Breadcrumb();
+        crumb.setMessage(msg);
+        crumb.setLevel(SentryLevel.INFO);
+
+        if(addStackTrace) {
+          StackTraceElement[] rawTrace = Thread.currentThread().getStackTrace();
+
+          // The call that gets us the stack-trace above... shows up in the
+          // stack trace, so we'll skip the first few (redundant) entries.
+          int skipCount = 3;
```

Comment:
Stack-trace generation produces the same trace as is viewable for Sentry exceptions, etc.  So, compare against the snippet below:

![image](https://github.com/keymanapp/keyman/assets/25213402/7db5e3a3-682d-4a73-92c2-a37de800b36a)

The breadcrumb-logging method is entry 3 within the stack trace, so we skip the first three entries.  Granted, it's _possible_ the count might differ a bit when on non-emulated devices, but even then, we should only skip the parts that we know make the message.